src/main/java/org/broadinstitute/hellbender/engine/ReadsDataSource.java

```diff
@@ -155,6 +161,16 @@ public ReadsDataSource( final List<Path> samPaths, final List<Path> samIndices,
         int samCount = 0;
         for ( final Path samPath : samPaths ) {
             // Ensure each file can be read
+            Function<SeekableByteChannel, SeekableByteChannel> wrapper = Function.identity();
+            Function<SeekableByteChannel, SeekableByteChannel> indexWrapper = Function.identity();
+            if (samPath.toUri().getScheme().equals("gs")) {
+                if (gcsWrapper != null) {
+                    wrapper = gcsWrapper;
+                }
+                if (cloudIndexWrapper != null) {
+                    indexWrapper = cloudIndexWrapper;
+                }
+            }
```

Comment:
Lines 164-173 could be collapsed into two lines using `? :` statements, which I think would help readability. Eg., 

    wrapper = BucketUtils.isCloudStorageUrl(samPath) && gcsWrapper != null ? gcsWrapper : Function.identity();

And similarly for the index.